src/pages/ListScholarships.jsx

```diff
@@ -41,6 +41,22 @@ function ListScholarships() {
   };
 
   const scrollTrigger = useScrollTrigger();
+  const [drawerHeight, setDrawerHeight] = useState(0);
+
+  useEffect(() => {
+    function calcHeight() {
+      setDrawerHeight(
+        window.innerHeight - ref.current?.getBoundingClientRect().top
+      );
+    }
+    window.addEventListener('scroll', calcHeight);
+    return () => {
+      window.removeEventListener('scroll', calcHeight);
+    };
+  }, [ref]);
```

Comment:
If you have the footer in a higher z dimension instead of this you could also just take the height of the footer and add a blank box at the bottom of the side bar with its height. That way you don't have to listen for scrolling and recalculate the height and rerender this component each time.

Reply:
Not sure how well this would work on mobile though.